src/Raven.Server/Documents/Queries/Results/TimeSeries/TimeSeriesRetriever.cs

```diff
@@ -1275,7 +1275,8 @@ public static DateTime ParseDateTime(string valueAsStr)
                 throw new ArgumentException($"Unable to parse timeseries from/to values. Got: {valueAsStr}{Environment.NewLine}" +
                                             $"The supported time formats are:{Environment.NewLine}" +
                                             $"{string.Join(Environment.NewLine, SupportedDateTimeFormats.OrderBy(f => f.Length))}");
-            return DateTime.SpecifyKind(date, DateTimeKind.Utc);
+
+            return TimeSeriesStorage.EnsureMillisecondsPrecision(date);
```

Comment:
Just to be absolutely sure - can it be a breaking change somehow? CC @karmeli87

Reply:
I don't think it is a breaking change, but we still need to keep `DateTime.SpecifyKind(date, DateTimeKind.Utc);`, no?

Reply:
Yep, done